lasy/utils/laser_utils.py

```diff
@@ -211,6 +206,78 @@ def get_full_field(laser, theta=0, slice=0, slice_axis="x", Nt=None):
     return env, ext
 
 
+def get_spectrum(grid, dim, bins=20, range=None, omega0=None, phase_unwrap_1d=None):
+    """
+    Get the the frequency spectrum of an envelope.
+
+    Parameters
+    ----------
+    grid : a Grid object.
+        It contains a ndarrays with the field data from which the
+        spectrum is computed, and the associated metadata. The last axis must
+        be the longitudinal dimension.
+        Can be the full electric field or the envelope.
+
+    dim : string (optional)
+        Dimensionality of the array. Only used if is_envelope is False.
+        Options are:
+
+        - 'xyt': The laser pulse is represented on a 3D grid:
+                 Cartesian (x,y) transversely, and temporal (t) longitudinally.
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    bins : int (optional)
+        Number of bins of the spectrum.
+
+    range : list of float (optional)
+        List of two values indicating the minimum and maximum frequency of the
+        spectrum.
+
+    omega0 : scalar
+        Angular frequency at which the envelope is defined.
+        Required if an only if is_envelope is True.
+
+    phase_unwrap_1d : boolean (optional)
+        Whether the phase unwrapping is done in 1D.
+        This is not recommended, as the unwrapping will not be accurate,
+        but it might be the only practical solution when dim is 'xyt'.
+
+    Returns
+    -------
+    spectrum : ndarray of doubles
+        Array with the spectrum amplitudes.
+
+    omega_spectrum : scalar
+        Array with the frequencies of the spectrum.
+    """
+    # Get frequency array.
```

Comment:
Maybe it's good to specify it at least once, we are slightly inaccurate in the wording.
```suggestion
    # Get the array of angular frequency.
```